### ISSUE TITLE
UnusedPrivateMember rule counts import declarations as "usage references"

### ISSUE THREAD

**Issue by luanpotter** [detekt/detekt#5941]
Note: this is a followup of https://github.com/detekt/detekt/pull/5935, adding yet another exception for the counting references rule.

## Expected Behavior

The following code:

```kotlin
import xyz.luan.Test

fun main() {
    val xyz = "a"
    val luan = "b"
    println("Hello, world!")
}
```

Should fail detekt validation of the `UnusedPrivateMember` for the 2 unused local variables.

## Observed Behavior

The code is not reported and detekt succeeds even though there are unused local variables.

Note: the same issue can happen with other types of unused private properties, like fields or parameters.

Any "bits" composing import statements are considered as variable uses.

## Steps to Reproduce

Please see [my repo](https://github.com/luanpotter/detekt-testbed/tree/luan.unused-private-property-improvements-2) for a fully reproducible example (just run `./gradlew detekt`). Or run detekt on the code snippets above.  Should be simple to reproduce. I am using version 1.22.0.

## Context

The fix for this one should be fairly simple, I will link [a PR to this issue with a proposal](https://github.com/detekt/detekt/pull/5942).

As mentioned, this is a followup of https://github.com/detekt/detekt/pull/5935
My idea is to chip away at some obvious edge cases to make the rule more correct, these two (packages and imports) being the most obvious.
Next potential scenarios on my mind are "right-hand side" of the "dot operator" references, though that might be a bit trickier (in the sense that it could cause false-positives which I think are even worse than these false-negatives).

## Your Environment

* Version of detekt used: 1.22.0
* Version of Gradle used: 8.0.2
* Gradle scan link: https://scans.gradle.com/s/snna4cxl7zbni
* Operating System and version: Arch Linux (kernel 22.3.0)
* Link to your project: https://github.com/luanpotter/detekt-testbed/tree/luan.unused-private-property-improvements-2
